Firestore/Example/FuzzTests/FuzzingTargets/FSTFuzzTestFieldPath.mm

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#include <cstddef>
+#include <cstdint>
+
+#import "Firestore/Example/FuzzTests/FuzzingTargets/FSTFuzzTestFieldPath.h"
+
+#import "Firestore/Source/Public/FIRFieldPath.h"
+
+namespace firebase {
+namespace firestore {
+namespace fuzzing {
+
+int FuzzTestFieldPath(const uint8_t *data, size_t size) {
+  @autoreleasepool {
+    // Convert the raw bytes to a string with UTF-8 format.
+    NSData *d = [NSData dataWithBytes:data length:size];
+    NSString *str = [[NSString alloc] initWithData:d encoding:NSUTF8StringEncoding];
+
+    // Fuzz test creating a FieldPath from an array with a single string.
+    @try {
+      NSArray *str_arr = [NSArray arrayWithObjects:str, nil];
+      [[FIRFieldPath alloc] initWithFields:str_arr];
+    } @catch (...) {
+      // Caught exceptions are ignored because they are not what we are after in
+      // fuzz testing.
+    }
+
+    // Split the string into an array using " .,/-" as separators.
+    @try {
+      NSCharacterSet *set = [NSCharacterSet characterSetWithCharactersInString:@" .,/_"];
+      NSArray *str_arr = [str componentsSeparatedByCharactersInSet:set];
+      [[FIRFieldPath alloc] initWithFields:str_arr];
+    } @catch (...) {
+      // Caught exceptions are ignored because they are not what we are after in
+      // fuzz testing.
+    }
+
+    // Try to parse the bytes as a string array and use it for initialization.
+    @try {
+      NSArray *str_arr = [NSKeyedUnarchiver unarchiveObjectWithData:d];
```

Comment:
I wonder, how many times is this case hit? Could you add a couple of counters, increment the first one right before the array is created, and the second one right after, and then compare their values? I just want to make sure `NSKeyedUnarchiver` works a reasonable amount of time.